src/tests/multihost/alltests/test_krb_fast_principal.py

```diff
@@ -0,0 +1,274 @@
+"""Automation for Krb Fast Principal tests ported from bash
+
+:requirement: krb_fast_prinicpal
+:casecomponent: sssd
+:subsystemteam: sst_idm_sssd
+:upstream: yes
+:status: approved
+"""
+import pytest
+from constants import ds_instance_name
+from sssd.testlib.common.utils import sssdTools
+
+
+def krb5_fast_setup(client, param1, param2, **kwargs):
+    """ To Customize domain parameter for Test Cases """
+    tools = sssdTools(client)
+    domain_section = f'domain/{ds_instance_name}'
+    domain_params = {
+        'krb5_use_fast': param1,
+        'krb5_fast_principal': param2}
+    for key, value in kwargs.items():
+        domain_params[key] = value
+    tools.sssd_conf(domain_section, domain_params)
+    tools.clear_sssd_cache()
+
+
+@pytest.fixture(scope='class')
+def custom_setup(session_multihost, setup_sssd_krb, create_posix_usersgroups, krb_connection_timeout):
+    """ Added neccessary sssd domain parameters """
+    tools = sssdTools(session_multihost.client[0])
+    sssd_params = {'services': "nss, pam",
+                   'config_file_version': 2}
+    tools.sssd_conf('sssd', sssd_params)
+    domain_section = f'domain/{ds_instance_name}'
+    domain_params = {'access_provider': 'krb5',
+                     'use_fully_qualified_names': 'False',
+                     'override_homedir': '/home/%u'}
+    tools.sssd_conf(domain_section, domain_params)
+    tools.clear_sssd_cache()
+
+
+@pytest.mark.tier2
+@pytest.mark.krbfastprinicpal
+@pytest.mark.usefixtures('custom_setup')
+class TestKrbFastPrincipal():
+    """
+    This is test case class for krb_fast_principal suite
+
+    Test FAST pre authentication by kerberos on the basis of krb5_fast_principal.
+    Test for valid and invalid principals, Test for principals with only
+    kerberos realm, Test for null principals, Tests for all the previous
+    scenarios with krb5_validate set to true.
+    """
+    @staticmethod
+    def test_0001_valid_principal(multihost):
```

Comment:
The tests are using krb5_fast_setup to change configuration, there should be a fixture backupsssdconf used for each test that does that so changes are reverted between/after the tests.